providers/os/resources/vulnmgmt.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"errors"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery/v9/llx"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/resources"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/upstream/gql"
+	"go.mondoo.com/cnquery/v9/providers/os/connection/shared"
+	mondoogql "go.mondoo.com/mondoo-go"
+)
+
+func (v *mqlVulnmgmt) lastAssessment() (*time.Time, error) {
+	mcc := v.MqlRuntime.Upstream
+	if mcc == nil || mcc.ApiEndpoint == "" {
+		return nil, resources.MissingUpstreamError{}
+	}
+
+	// get new gql client
+	mondooClient, err := gql.NewClient(mcc.UpstreamConfig, mcc.HttpClient)
+	if err != nil {
+		return nil, err
+	}
+
+	if v.MqlRuntime.Upstream.AssetMrn == "" {
+		return nil, errors.New("no asset mrn available")
+	}
+	lastUpdate, err := mondooClient.LastAssessment(v.MqlRuntime.Upstream.AssetMrn)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debug().Str("time", lastUpdate).Msg("search for package last update")
+	if lastUpdate == "" {
+		return nil, errors.New("no update time available")
+	}
+
+	var lastUpdateTime *time.Time
+	if lastUpdate != "" {
+		parsedLastUpdateTime, err := time.Parse(time.RFC3339, lastUpdate)
+		if err != nil {
+			return nil, errors.New("could not parse last update time: " + lastUpdate)
+		}
+		lastUpdateTime = &parsedLastUpdateTime
+	} else {
+		lastUpdateTime = &llx.NeverFutureTime
+	}
+
+	return lastUpdateTime, nil
+}
+
+func (v *mqlVulnmgmt) cves() ([]interface{}, error) {
+	return nil, v.populateData()
+}
+
+func (v *mqlVulnmgmt) advisories() ([]interface{}, error) {
+	return nil, v.populateData()
+}
+
+func (v *mqlVulnmgmt) packages() ([]interface{}, error) {
+	return nil, v.populateData()
+}
+
+func (v *mqlVulnmgmt) stats() (*mqlAuditCvss, error) {
+	return nil, v.populateData()
```

Comment:
shouldn't that return also the actual value?

Reply:
Not needed. That's some MQL black magic: https://github.com/mondoohq/cnquery/blob/main/providers/os/resources/command.go#L58

But a comment might also be helpful here. I added it.